### ISSUE TITLE
Undefined Failure | Abstract Consumer

### ISSUE THREAD

**Issue by basdag** [evansolomon/nodejs-kinesis-client-library#7]
Hi again @evansolomon .

There is a point were the library has processed all the data from a closed stream and the conditional is failing since the Kinesis Client is not returning anything in data.Records.

```
/src/node_modules/aws-sdk/lib/request.js:32
          throw err;
                ^
TypeError: Cannot read property 'length' of undefined
    at Response.<anonymous> (/src/node_modules/kinesis-client-library/AbstractConsumer.js:227:55)
    at Request.<anonymous> (/src/node_modules/aws-sdk/lib/request.js:350:18)
    at Request.callListeners (/src/node_modules/aws-sdk/lib/sequential_executor.js:100:18)
    at Request.emit (/src/node_modules/aws-sdk/lib/sequential_executor.js:77:10)
    at Request.emit (/src/node_modules/aws-sdk/lib/request.js:604:14)
    at Request.transition (/src/node_modules/aws-sdk/lib/request.js:21:12)
    at AcceptorStateMachine.runTo (/src/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /src/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/src/node_modules/aws-sdk/lib/request.js:22:9)
    at Request.<anonymous> (/src/node_modules/aws-sdk/lib/request.js:606:12)
```

I would do a PR when I get a chance, otherwise the line to change is 

```
if (data.NextShardIterator == null && data.Records.length === 0) {
    return _this._markFinished()
}
```

to

```
if (null === data.NextShardIterator && (!data.Records || 0 === data.Records.length)) {
    return _this._markFinished()
}
```

Thanks

**Comment by evansolomon** [comment]
Interesting.  I'm pretty sure this violates Amazon's API documentation, which says that `Records` will always be an array.

**Comment by evansolomon** [comment]
Made the change you suggested and published it as version 0.2.3
